Src/FluentAssertions/Primitives/StringAssertions.cs

```diff
@@ -387,6 +387,16 @@ public AndConstraint<TAssertions> NotMatch(string wildcardPattern, string becaus
             return new AndConstraint<TAssertions>((TAssertions)this);
         }
 
+        public AndConstraint<TAssertions> MatchRegex([RegexPattern][StringSyntax("Regex")] string regularExpression,
+            OccurrenceConstraint occurrenceConstraint, string because = "", params object[] becauseArgs)
+        {
```

Comment:
The two new overloads need to match closer to the existing `MatchRegex` methods in terms of input validation.

Reply:
Ok, do you mean that `Guard.ThrowIfArumentNull` and the `try/catch` block? Is that not considered as code duplication, because it would be copy/paste.

Reply:
Yes, these are the parts I mean.

I wouldn't consider copying that as code duplication.

Reply:
Oh.. ok then I will treat this two overloads as "independant" from the other two `MatchRegex()` methods?

Reply:
A possible solution could also be to implement a new `Guad` method e.g. `Guard.ThrowIfRegexIsInvalid()` to make the regex testing in one spot...